stock_picking_auto_create_lot_biochousa/models/stock_picking.py

```diff
@@ -0,0 +1,57 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models, _
+from datetime import datetime
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def button_validate(self):
+
+        if self.picking_type_id.auto_create_lot:
+            for line in self.move_line_ids.filtered(
+                lambda x: (
+                    not x.lot_id
+                    and not x.lot_name
+                    and x.product_id.tracking != "none"
+                    and x.product_id.auto_create_lot
+                )
+            ):
+                # Create lot_name as Id.Product+Contact/Warehouse code+Date+1/0(ECO)
+                # Code Product = Stock_move_line.product_id --> Product_product.default_code
+                if line.product_id.default_code:
+                    product_code = line.product_id.default_code
+                else:
+                    raise UserError(_('Cannot find the internal reference of product.'))
+
+                # If exist: PartnerRef = stock_picking.partner_id --> res_partner.ref
+                # Else: PartnerRef = stock_warehouse.code
+                if self.partner_id and self.partner_id.ref:
+                    parter_code = self.partner_id.ref
+                else:
+                    parter_code = self.location_dest_id.get_warehouse().code
+                    
+                # Current date format YYYYMMDD
+                str_date = datetime.today().strftime('%Y%m%d')
+                
+                # Search value of attribute ECO
+                str_bio = "0"
+                eco_attr_line = line.product_id.attribute_line_ids.filtered(lambda x: x.attribute_id.name == "ECO")
+                if eco_attr_line:
+                    if not (eco_attr_line.value_ids[0].name == "No"):
+                        str_bio = "1"
+                else:
+                    raise UserError(_('Cannot find the ECO attribute of product.'))
```

Comment:
@lmiguens-solvos If a product has no ECO attribute, should we fire an exception, or simple set `str_bio` to `0`?

Reply:
With this fix it will always have a value of 1 unless the product is related to the partner, which would then be 0.